pysatCDAAC/instruments/cosmic_gps.py

```diff
@@ -164,8 +164,8 @@ def list_files(tag=None, sat_id=None, data_path=None, format_str=None):
         # due to multiple spacecraft and antennas
         # this ensures that we can make the times all unique for the file list
         idx = np.argsort(uts)
-        # adding linearly increasing offsets less than 0.01 s
-        shift_uts = np.mod(np.arange(len(year)), 1E3) * 1.E-5 + 1.E-5
+        # adding linearly increasing offsets less than 0.0001 s
+        shift_uts = np.mod(np.arange(len(year)), 1E3) * 1.E-7 + 1.E-7
```

Comment:
```suggestion
        # adding linearly increasing offsets less than 0.01 s
        shift_uts = np.mod(np.arange(len(year)), 1E4) * 1.E-6 + 1.E-6
```
I ran this on data from 2009 - 2014. The issue is multiple files in a single day that will reduce to a common time. By my check there are less than 10,000 files per day, hence the switch in mod. Time increments can only go so small for normal date time objects. hence the switch from the proposed 1E-7 to 1E-6.

Reply:
Does this change work on your system as well?

Reply:
That's working.  Thanks!